File: ResiBuy.Server/Infrastructure/Model/Product.cs
namespace ResiBuy.Server.Infrastructure.Model
{
    public class Product
    {
        public int                   Id           { get; set; }
        public string                 Name         { get; set; }
        public string                 Describe     { get; set; }
        public bool                   IsOutOfStock { get; set; }
        public int?                    PromotionId     { get; set; }
        public DateTime               CreatedAt    { get; set; }
        public DateTime               UpdatedAt    { get; set; }
        public Guid                   StoreId      { get; set; }
        public Guid                   CategoryId   { get; set; }
        public DateTime?              ExpiryDate { get; set; }
        public int?                   WarrantyMonths { get; set; } // in months
        public Store                  Store        { get; set; }
        public Category               Category     { get; set; }
        public List<ProductDetail> ProductDetails { get; set; }
        public Promotion Promotion { get; set; }

        public Product()
        {
            
        }

        public Product(
           string name,
           string describe,
           int promotionId,
           Guid storeId,
           Guid categoryId,
           bool isOutOfStock = false)

        {
            Name = name;
            Describe = describe;
            PromotionId = promotionId;
            StoreId = storeId;
            CategoryId = categoryId;
            IsOutOfStock = isOutOfStock;
            CreatedAt = DateTime.UtcNow;
            UpdatedAt = DateTime.UtcNow;
            ProductDetails = new List<ProductDetail>();
           
        }

        public void UpdateProduct(
              string name,
              string describe,
              int promotionId,
              Guid categoryId,
              bool isOutOfStock)
        {
            Name = name;
            Describe = describe;
            PromotionId = promotionId;
            CategoryId = categoryId;
            IsOutOfStock = isOutOfStock;
            UpdatedAt = DateTime.UtcNow;

        }

        public void UpdateStatusProduct(bool isOutOfStock)
        {
            IsOutOfStock = isOutOfStock;
        }
    }


}

File: resibuy.client/src/types/product.ts
export interface ImageDto {
  id: string;
  url: string;
  thumbUrl: string;
  name: string;
}

export interface AdditionalDataDto {
  id: number;
  key: string;
  value: string;
}

export interface ProductDetailDto {
  id: number;
  isOutOfStock: boolean;
  productId: number;
  sold: number;
  price: number;
  weight: number;
  quantity: number;
  image: ImageDto;
  additionalData: AdditionalDataDto[];
  // Không cần cartItems, orderItems, reviews ở DTO phía client
}

export interface CategoryDto {
  id: string;
  name: string;
  status: boolean;
}

export interface PromotionDto {
  id: number;
  name: string;
  discount: number;
  startDate: string;
  endDate: string;
  isActive: boolean;
}

export interface ProductDto {
  id: number;
  name: string;
  describe: string;
  isOutOfStock: boolean;
  promotionId: number;
  createdAt: string;
  updatedAt: string;
  storeId: string;
  categoryId: string;
  category: CategoryDto;
  promotion: PromotionDto;
  productDetails: ProductDetailDto[];
}

export interface ProductFilter {
  pageNumber: number;
  pageSize: number;
  categoryId?: string;
}
